client/blocks/jitm/templates/sidebar-banner.jsx

```diff
@@ -1,15 +1,57 @@
 /**
- * External Dependencies
+ * External dependencies
  */
 import React from 'react';
 import Gridicon from 'components/gridicon';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import UpsellNudge from 'blocks/upsell-nudge';
+import { abtest } from 'lib/abtest';
 
 /**
  * Style dependencies
  */
 import './sidebar-banner.scss';
 
-export default function SidebarBannerTemplate( { icon, CTA, message, onClick, trackImpression } ) {
+export default function SidebarBannerTemplate( {
+	icon,
+	CTA,
+	message,
+	onClick,
+	trackImpression,
+	id,
+	event,
+	featureClass,
+	tracks,
+	...props
+} ) {
+	if ( abtest( 'sidebarUpsellNudgeUnification' ) === 'variantShowUnifiedUpsells' ) {
+		const clickName = get( tracks, [ 'click', 'name' ] ) || `jitm_nudge_click_${ id }`;
+		const clickProps = get( tracks, [ 'click', 'props' ] );
+		const displayName = get( tracks, [ 'display', 'name' ] ) || `jitm_nudge_impression_${ id }`;
+		const displayProps = get( tracks, [ 'display', 'props' ] );
+		const jitmProps = { id: id, jitm: true };
+
+		return (
+			<UpsellNudge
+				callToAction={ CTA.message }
+				compact
+				event={ displayName }
+				href={ CTA.link }
+				onClick={ onClick }
```

Comment:
🚨Passing both the onClick and tracksNames will cause a double tracks event

<img width="1185" alt="Screen Shot 2020-01-23 at 3 05 14 PM" src="https://user-images.githubusercontent.com/1270189/73031733-c8976880-3df1-11ea-8a13-b7963a008cc0.png">

In this case, passing through onClick would probably have equivalent behavior, so I'd recommend removing the pass through on the tracks* properties.

Reply:
I think we can remove `onClick`, since the only active nudges using it are JITMs, and they're only tracking click events. (I converted the non-JITMs to use `UpsellNudge` tracking as well).

My thought was, `UpsellNudge` would handle all impression, click, and dismiss events, so everything happens in one component. I'm second-guessing myself now, though. :D Would it be better to let every nudge set its own event functions and pass them through via `TrackImpression`, `onClick` and `onDismiss`? That seems more flexible, but potentially messier to keep track of.

Reply:
In this case keeping `onClick` would preserve existing behavior, while I'd expect new UpsellNudge usages **not to be firing tracks events** in onClick (it'd be some other action like display a modal). I don't have super strong feelings here though, so feel free to take your pick of keeping onClick here or the tracks* props.